pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -160,30 +131,17 @@ func (t *ResolvedPipelineRunTask) conditionsSkip() bool {
 	return false
 }
 
-func (t *ResolvedPipelineRunTask) whenExpressionsSkip(facts *PipelineRunFacts) bool {
-	if t.checkParentsDone(facts) {
-		if len(t.PipelineTask.WhenExpressions) > 0 {
-			if !t.PipelineTask.WhenExpressions.HaveVariables() {
-				if !t.PipelineTask.WhenExpressions.AllowsExecution() {
-					return true
-				}
+func (t *ResolvedPipelineRunTask) whenExpressionsSkip() bool {
+	if len(t.PipelineTask.WhenExpressions) > 0 {
```

Comment:
Not blocking: Was there an explicit style decision involved in making this nested `if`s instead of `&&`?

Reply:
Not really, here I only removed the outer `if`, so it wasn't really a decision :)
I tried to minimise changes to the code in case we want to use this for minor releases.